portal-web/docroot/html/taglib/ui/page_iterator/start.jsp

```diff
@@ -410,6 +412,86 @@ NumberFormat numberFormat = NumberFormat.getNumberInstance(locale);
 	}
 </script>
 
+<script data-senna-track="temporary" type="text/javascript">
+
+	var dropdown = document.getElementById("<%= namespace %>");
+	var button = dropdown.querySelector('.dropdown-toggle');
+	var list = dropdown.querySelector('.dropdown-menu');
+	var options = list.querySelectorAll('.dropdown-item');
+	var selectedItemValue = button.dataset.attribute;
+
+	function onButtonKeyDown(event) {
+		if (event.key === 'ArrowDown' || event.key === 'ArrowUp' || event.key === 'Enter' || event.key === 'Space') {
+			event.preventDefault();
+			list.classList.add('show');
+			button.setAttribute('aria-expanded', 'true');
+			var selectedOption = null;
+
+			Array.from(options).find(function(option) {
+				if (option.id == selectedItemValue) {
+					selectedOption = option;
+				}
+			});
+
+			if (selectedOption) {
+				selectedOption.focus();
+			}
+		}
+	}
+
+	button.addEventListener('keydown',onButtonKeyDown );
+
+	var destroyDropDownPagination = function () {
+		button.removeEventListener('keydown', onButtonKeyDown);
+
+		Liferay.detach('beforeScreenFlip', destroyDropDownPagination);
+	};
+
+	Liferay.on('beforeScreenFlip', destroyDropDownPagination);
+
+	options.forEach(function(option, index) {
+
+		function handleKeyEvents(event) {
+			if (event.key === 'ArrowDown') {
+				event.preventDefault();
+
+				if (index < options.length - 1) {
+					options[index + 1].focus();
+				}
+			} else if (event.key === 'ArrowUp') {
+				event.preventDefault();
+
+				if (index > 0) {
+					options[index - 1].focus();
+				}
+			}
+		}
+
+		option.addEventListener('keydown', handleKeyEvents);
+
+		var destroyKeyEvents = function () {
+			option.removeEventListener('keydown', handleKeyEvents);
+
+			Liferay.detach('beforeScreenFlip', destroyKeyEvents);
+		};
+
+		Liferay.on('beforeScreenFlip', destroyKeyEvents);
```

Comment:
You can do this a little better here to avoid this being declared within the iteration. For example you can move `handleKeyEvents` out of forEach along with `onButtonKeyDown` instead of declaring a new event to remove them you can do this just once in `destroyDropDownPagination`.

```js
function handleKeyEvents(event) {}

options.forEach(function(option) {
  option.addEventListener('keydown', handleKeyEvents);
});

var destroyDropDownPagination = function () {
	button.removeEventListener('keydown', onButtonKeyDown);
  options.forEach(function(option) {
  	option.removeEventListener('keydown', handleKeyEvents);
  });

	Liferay.detach('beforeScreenFlip', destroyDropDownPagination);
};
Liferay.on('beforeScreenFlip', destroyDropDownPagination);
```

Reply:
Thanks @matuzalemsteles! All fixed :smile: